### DISCUSSION TITLE
How to improve verbosity when navigating in / out of containers on the web

### DISCUSSION THREAD

**Discussion by Qchristensen** [nvaccess/nvda#16626] [Polls]
Initial discussion started on Mastodon at: https://fosstodon.org/@amir@tweesecake.social/112509353851073904

Summarised in: https://aus.social/@jcsteh/112518625440837421 by Jamie Teh, I've turned them into poll options below.  Please vote and add any extra discussion below.

Additional notes from Jamie in his post:
But support for #3 (put starts and ends of containers on separate lines like Jaws) is already pretty polarised between users just in the original Mastodon thread.

#4 (report container stars and ends) would result in "ends" being reported at the end of the last line of the container when moving down rather than the start of the first line of the next container (issue 10446).  This just shifts the verbosity rather than eliminating it, but it might still solve the problem for some users.

#5 (optionally don't report container exits at all) though I feel like this might end up causing more confusion than it fixes."

<Note from Quentin: My thought was that anything implemented from this would be an option in addition to the current behaviour, but since it's also worth knowing how many respondents prefer the current behaviour, I've added that as an option as well>

**Comment by jscholes** [comment]
My personal view is that the complaints around verbosity are completely valid and understandable, particularly the idea of being overwhelmed by multiple announcements, including the subsequent content outside of a container.  However, I think there is significant risk here that in the pursuit of a "better" option for some users, the experience may be degraded for others.  For instance:
1. A message like "out of five containers", while admittedly concise, requires users to understand what a container is.  I'm also unclear what the user experience would be in the opposite direction (i.e. moving into multiple containers), and for determining what those container types actually were for people who legitimately wanted or needed to know that information.
2. "out of three tables, two block quotes": this raises questions in my mind of ordering, e.g. were the tables inside a block quote, or is it just indiscriminate digestion?  If the latter, I have the same questions as for `#1`.  It also raises red flags around cognitive load every time users are prompted to count stuff they didn't have to before.
3. Spreading out starts and ends of containers to separate browse mode lines could significantly increase the number of repetitive keypresses needed to navigate a page which are already hard on users' wrists, e.g. Down Arrow.
4. "Report container starts and ends, rather than entry and exit": I understand the distinction, but not how this would solve the problem.  "out of list out of list out of list out of list", and "end of list end of list end of list end of list", seem to represent the same levels of complexity and verbosity to me (and are the same number of syllables).
5. Not reporting container exists at all seems like a non-starter for people who legitimately want (or need) to understand the structure of a page, e.g. for maximum efficiency in navigating, or for professional accessibility testing reasons.

I don't view any of these as blockers to making the functionality better for people who struggle with the current approach.  I think some of them show a lot of promise, but I think configurability is the key.  This will understandably make implementation harder and slower, but hopefully my previous comments make it clear why I don't view this as an easy problem to solve regardless, even with the options being on the table that we have.

**Reply by Qchristensen** [reply]
Good point James - my thought had been that anything implemented here would be an option so you could still have the current functionality.  I've just updated the poll to hopefully clarify that.

**Comment by CyrilleB79** [comment]
I have other suggestions than the ones proposed that may already improve the current design, and maybe not so hard to implement:
1. A bit as proposal `#1`, but more precisely: condense two or more successive reporting of the same container, both for entry and exit. E.g: "out of table, out of list, out of list, out of table, block quote, block quote" is reduced to 
"out of table, out of 2 lists, out of table, in 2 block quotes".
2. Other condensation rules such as "out of list, list" becomes "new list"
3. Only report the last exited type of container. E.g. "out of table, out of list, out of list, out of table" becomes "out of 2 tables". We expect that the user remembers that the two nested lists were content of the outer table.
4. Add comma to add a pause between two structure reports. E.g. instead of "out of list out of list", you get "out of list, out of list". Does not reduce complexity but avoids to be bombed by a somewhat unstructured container information.

To be discussed...

Note: I have not voted in the poll because since we are talking of optional configuration, there is no point in chosing only one option.
